flink-scala/src/main/scala/org/apache/flink/api/scala/DataSet.scala

```diff
@@ -178,6 +178,60 @@ class DataSet[T: ClassTag](set: JavaDataSet[T]) {
   }
 
   /**
+   * Sets the minimum and maximum resources of this operation.
+   */
+  def setResource(minResource: ResourceSpec, maxResource: ResourceSpec) = {
+    javaSet match {
+      case ds: DataSource[_] => ds.setResource(minResource, maxResource)
+      case op: Operator[_, _] => op.setResource(minResource, maxResource)
+      case di: DeltaIterationResultSet[_, _] =>
+        di.getIterationHead.setResource(minResource, maxResource)
+      case _ =>
+        throw new UnsupportedOperationException("Operator " + javaSet.toString + " cannot have " +
```

Comment:
I would suggest to change the error message in all places to something like "Operator <op> does not support configuring custom resources specs".